cellmlmanip/parser.py

```diff
@@ -171,14 +171,27 @@ def _add_components(self, model):
         # for each component defined in the model
         for element in component_elements:
             # component are only kept in parser to resolve relationships and connections
-            self.components[element.get('name')] = _Component(element.get('name'))
+            name = element.get('name')
+            self.components[name] = _Component(name)
 
             # process the <variable> tags in this component
             variable_to_symbol = self._add_variables(element)
 
             # process the <math> tags in this component
             self._add_maths(element, variable_to_symbol)
 
+            # Raise error if component units are defined
+            component_units = element.findall(Parser.with_ns(XmlNs.CELLML, 'units'))
+            if component_units:
+                raise RuntimeError(
```

Comment:
We seem to be using `ValueError` for other parser errors, so probably best to be consistent.